### ISSUE TITLE
Check for regular passwords

### ISSUE THREAD

**Issue by niccokunzmann** [CoderDojo/community-platform#833]
There are lists online which contain the most regular passwords.
When entering a new password, one can check that it is contained.

**Comment by Wardormeur** [comment]
@niccokunzmann The idea would be to check against common passwords to enforce security by disabling those common ones? We already enforce at least (a number OR a special char) and a length of 8.
@butlerx What's your opinion on this?

**Comment by niccokunzmann** [comment]
@Wardormeur Yes, this is the idea. I would argue if there are passwords with a number or a special character among those most popular passwords, the enforcement does not help in those cases.
I found this website. http://www.passwordrandom.com/most-popular-passwords 12345678 is among them There should be a list somewhere on the web.

**Comment by niccokunzmann** [comment]
Found one: https://raw.githubusercontent.com/danielmiessler/SecLists/master/Passwords/10_million_password_list_top_1000000.txt

**Comment by Wardormeur** [comment]
Interesting to see it comes from the OWASP projejct . :+1: 
I'm a bit concerned in term of priority:  as much as I'd want security to be a priority, this one is about prevention of creation of a weak password, considering we already have a unique salt per password.
Any recommendation for the user message to display in case of weak password?
"This password has been recognized to be used as one of the 10 000 most common password, please change it to a more secure one or use a password manager (LastPass, PassPack)" ?

**Comment by butlerx** [comment]
id agree with @niccokunzmann that you'd be unlikely to see any password with special char on that list but youd probably find some mix of numbers and letter password.
i think its a good idea, but we need to be careful of removing possible passwords as doing that means the pool of passwords a attacker would have to test is smaller

**Comment by Wardormeur** [comment]
I've asked on twitter a couple of people and it seems it's a decent recommendation
ref (https://pages.nist.gov/800-63-3/sp800-63b.html) : 
`When processing requests to establish and change memorized secrets, verifiers SHALL compare the prospective secrets against a list that contains values known to be commonly-used, expected, or compromised. For example, the list MAY include (but is not limited to):

Passwords obtained from previous breach corpuses.
Dictionary words.
Repetitive or sequential characters (e.g. ‘aaaaaa’, ‘1234abcd’).
Context specific words, such as the name of the service, the username, and derivatives thereof.`
So definitely something we should do, thanks @niccokunzmann  :+1:
